subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/configuration/ImpactAnalysisPolicy.kt

```diff
@@ -1,33 +1,17 @@
 package com.avito.instrumentation.configuration
 
-import com.avito.instrumentation.impact.AnalyzeTestImpactTask
-import com.avito.instrumentation.impact.analyzeTestImpactTask
-import org.gradle.api.Project
-import org.gradle.api.file.RegularFile
-import org.gradle.api.provider.Provider
-import org.gradle.api.tasks.TaskProvider
 import java.io.Serializable
 
-@Suppress("UnstableApiUsage")
 sealed class ImpactAnalysisPolicy : Serializable {
 
     sealed class On : ImpactAnalysisPolicy() {
 
-        abstract fun getArtifact(task: AnalyzeTestImpactTask): Provider<RegularFile>
+        object RunAffectedTests : On()
 
-        fun getTask(project: Project): TaskProvider<AnalyzeTestImpactTask> = project.tasks.analyzeTestImpactTask()
+        // Use RunModifiedTests, it also includes new tests
+        object RunNewTests : On()
 
-        object RunAffectedTests : On() {
-            override fun getArtifact(task: AnalyzeTestImpactTask): Provider<RegularFile> = task.testsToRunFile
-        }
-
-        object RunNewTests : On() {
-            override fun getArtifact(task: AnalyzeTestImpactTask): Provider<RegularFile> = task.addedTestsFile
-        }
-
-        object RunModifiedTests : On() {
-            override fun getArtifact(task: AnalyzeTestImpactTask): Provider<RegularFile> = task.modifiedTestsFile
-        }
```

Comment:
Will it break backward compatibility? Is it expensive to do so?

Reply:
it will technically, but getArtifact should be internal; and it was used only in plugin